yt_dlp/extractor/yappy.py

```diff
@@ -0,0 +1,97 @@
+            'like_count': int,
+            'description': 'Опиши свой характер 3 словами🙃\n#психология #дружба #отношения',
+            'uploader_id': '6793ee3581974a3586fc01e157de6c99',
+            'view_count': int,
+            'repost_count': int,
+            'uploader': 'LENA SHTURMAN',
+            'upload_date': '20230126',
+            'thumbnail': 'https://cdn-st.ritm.media/static/pic/user_thumbnails/6e76bb4bbad640b6/9ec84c115b2b1967/1674716171.jpg',
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        json_ld = self._search_json_ld(webpage, video_id)
+        nextjs_data = self._search_nextjs_data(webpage, video_id)
+
+        media_data = (
+            traverse_obj(
+                nextjs_data, ('props', 'pageProps', ('data', 'OpenGraphParameters')), get_all=False)
+            or self._download_json(f'https://yappy.media/api/video/{video_id}', video_id))
+
+        media_url = media_data['link']
+        has_watermark = str(media_url).endswith('wm.mp4')
+
+        formats = [{
+            'url': url_or_none(media_url),
+            'ext': 'mp4',
+            'format_note': 'Watermarked' if has_watermark else None,
+            'preference': -10 if has_watermark else None
+        }]
+
+        if has_watermark and media_url:
+            formats.append({
+                'url': url_or_none(str(media_url).replace('-wm.mp4', '.mp4')),
+                'ext': 'mp4'
+            })
+
+        if media_data.get('audio'):
+            formats.append({
+                'url': traverse_obj(media_data, ('audio', 'link')),
+                'ext': 'mp3',
+                'vcodec': 'none'
+            })
+
+        return {
+            'id': video_id,
+            'title': (json_ld.get('description') or self._html_search_meta(['og:title'], webpage)
+                      or self._html_extract_title(webpage)),
+            'formats': formats,
+            'thumbnail': (media_data.get('thumbnail')
+                          or self._html_search_meta(['og:image', 'og:image:secure_url'], webpage)),
+            'description': (media_data.get('description') or json_ld.get('description')
+                            or self._html_search_meta(['description', 'og:description'], webpage)),
+            'timestamp': unified_timestamp(media_data.get('publishedAt') or json_ld.get('timestamp')),
+            'view_count': int_or_none(media_data.get('viewsCount') or json_ld.get('view_count')),
+            'like_count': int_or_none(media_data.get('likesCount')),
+            'uploader': traverse_obj(media_data, ('creator', 'firstName')),
+            'uploader_id': traverse_obj(media_data, ('creator', ('uuid', 'nickname')), get_all=False),
+            'categories': traverse_obj(media_data, ('categories', ..., 'name')) or None,
+            'repost_count': int_or_none(media_data.get('sharingCount'))
+        }
```

Comment:
Instead of taking out specific keys, use `merge_dict` with `json_ld`

Reply:
I take `json_ld` key in specific keys to set extraction priority correctly, some of `json_ld` keys are not accurate.